formatting.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: page
+title: Formatting Flutter code
+sidebar: home_sidebar
+permalink: /formatting/
+---
+
+Flutter source code style guidelines, and support for automatic formatting.
+
+* TOC Placeholder
+{:toc}
+
+## Flutter formatting guidelines
+
+Flutter framework code and sample code uses a consistent style as detailed in
+our [style
+guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo).
+
+Flutter application developers are welcome to follow this style as well, but
+this is by not required. Flutter will work regardless of what style is used
+when writing applications.
+
+## Automatically formatting code
+
+While your code may follow any preferred style -- in our experience -- teams of
+developers may find it more productive to:
+
+1. have a single, shared style, and
+
+1. to enforce this style through automatic formatting.
+
+The alternative is often tiring formatting debates during code reviews, where
+time may be better spent on code behavior rather than code style. (*Note*:
+Automatically formatting code may in some hard cases lead to different results
```

Comment:
I think we can drop this parens. It's an interesting link but not in the critical path to enabling our developers to use the formatter.